frontend/src/lib/components/IntervalFilter/IntervalFilter.tsx

```diff
@@ -1,19 +1,40 @@
 import { intervalFilterLogic } from './intervalFilterLogic'
 import { useActions, useValues } from 'kea'
 import { intervals } from 'lib/components/IntervalFilter/intervals'
-import { InsightType, IntervalType } from '~/types'
+import { BaseMathType, InsightType, IntervalType } from '~/types'
 import { insightLogic } from 'scenes/insights/insightLogic'
-import { LemonSelect } from '@posthog/lemon-ui'
+import { LemonSelect, lemonToast } from '@posthog/lemon-ui'
 
 interface InvertalFilterProps {
     view: InsightType
     disabled?: boolean
+    activeUsersMath?: BaseMathType.WeeklyActiveUsers | BaseMathType.MonthlyActiveUsers | null
 }
 
-export function IntervalFilter({ disabled }: InvertalFilterProps): JSX.Element {
+export function IntervalFilter({ disabled, activeUsersMath }: InvertalFilterProps): JSX.Element {
     const { insightProps } = useValues(insightLogic)
     const { interval } = useValues(intervalFilterLogic(insightProps))
     const { setInterval } = useActions(intervalFilterLogic(insightProps))
+
+    const enabledIntervals: Record<string, any> = { ...intervals }
+
+    if (activeUsersMath) {
+        // Disallow grouping by hour for WAUs/MAUs as it's an expensive query that produces a view that's not useful for users
+        delete enabledIntervals.hour
```

Comment:
Hmm, this works, but I worry users might be confused _why_ for their particular insight half of the options is poof (especially if you have multiple series and only one is weekly/monthly active users). Can we use `disabledReason` instead? 
Extra suggestion: This is logic… and Kea logics are great for separating logic from presentation. This would be great as an `intervals` selector in `intervalFilterLogic`.